docs/_source/tutorials_and_integrations/tutorials/feedback/end2end_examples/add-metadata-003.ipynb

```diff
@@ -0,0 +1,637 @@
+   "metadata": {},
+   "source": [
+    "Note that the `name` will be the name of the metadata property shown internally. `title` is again the name of the metadata property displayed in the UI. `values` will be the list of possible values for the metadata property. To add it to the dataset, we can simply use the `add_metadata_property` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.add_metadata_property(terms_metadata_property)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us add the metadata to individual records as well. For demonstration purposes here, we can create a list of groups in randomly select one to assign to each record. As we normally do, we should be careful to divide records into groups in equal numbers.\n",
+    "\n",
+    "We firstly learn how many records we have in the dataset and how many groups we have defined in our `TermsMetadataProperty`. Then, we create a list of group names at the number of records. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "length_dataset = len(dataset)\n",
+    "num_groups = len(terms_metadata_property.values)\n",
+    "\n",
+    "groups = [f\"group-{i}\" for i in range(1, num_groups + 1) for _ in range(length_dataset // num_groups + (1 if i <= length_dataset % num_groups else 0))]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we choose one group name randomly, assign it to the record and remove it from the list.\n",
+    "\n",
+    "The adding process slightly differs among the `FeedbackDataset` and `RemoteFeedbackDataset`. For the former, we can simply add metadata by assigning the value. For the latter, we should use the `update_records` method. Let us see how we can do it for both."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# For the local dataset\n",
+    "for record in dataset.records:\n",
+    "    random_record = random.choice(groups)\n",
+    "    record.metadata[\"group\"] = random_record\n",
+    "    groups.remove(random_record)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# For the remote dataset\n",
+    "modified_records = [record for record in dataset.records]\n",
+    "\n",
+    "for record in modified_records:\n",
+    "    random_record = random.choice(groups)\n",
+    "    record.metadata[\"group\"] = random_record\n",
+    "    groups.remove(random_record)\n",
+    "\n",
+    "dataset.update_records(modified_records)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now see that the metadata property is added to the dataset and we can filter our records accordingly."
```

Comment:
maybe here we can show a picture, or if it's seen in the last one just reference it?

Reply:
Actually, there is one in the end so it might be better to remove this at all and show all properties in the picture below.